lib/active_model/serializer/utils.rb

```diff
@@ -0,0 +1,25 @@
+module ActiveModel::Serializer::Utils
+  module_function
+
+  # Gives the existing lookup paths that are suffixes of the fully qualified path of `klass`.
+  # @param [Object] klass
+  # @return [Array<Symbol>]
+  def nested_lookup_paths(klass)
+    paths = klass.to_s.split('::').reverse.reduce([]) { |a, e| a + [[e] + Array(a.last)] }.reverse
```

Comment:
So that

```ruby
          @share1 = Share.new
TweetSerializer.serializer_for(@share1) #=> TweetSerializer::ShareSerializer
ActiveModel::Serializer.serializer_for(@share1) #=> ShareSerializer
```